src/quacc/atoms/skzcam.py

```diff
@@ -23,13 +26,751 @@
+        mp2_oniom3_hl
+            A dictionary containing the information for the "high-level" MP2 treatment of ONIOM layer 3.
+        mp2_oniom4_hl
+            A dictionary containing the information for the "high-level" MP2 treatment of ONIOM layer 4.
+        deltaCC
+            A dictionary containing the information for the deltaCC calculation.
+
+        Returns
+        -------
+        None
+        """
+
+        def _format_oniom_dict(oniom_dict: SKZCAMInfo) -> SKZCAMInfo:
+            """
+            Format the oniom dictionary to include the default values for the SKZCAM calculations if the user hasn't specified it themselves.
+
+            Parameters
+            ----------
+            oniom_dict
+                The dictionary containing the information for the SKZCAM calculation.
+
+            Returns
+            -------
+            SKZCAMInfo
+                The formatted dictionary containing the information for the SKZCAM calculation.
+
+            """
+
+            # Raise error if the code is not provided or not specified as 'mrcc' or 'orca'.
+            if "code" not in oniom_dict:
+                raise ValueError("The code must be specified.")
+            if oniom_dict["code"] not in ["mrcc", "orca"]:
+                raise ValueError("The code must be either 'mrcc' or 'orca'.")
+
+            if "multiplicities" not in oniom_dict:
+                oniom_dict["multiplicities"] = {
+                    "adsorbate_slab": 1,
+                    "adsorbate": 1,
+                    "slab": 1,
+                }
+            if "ecp" not in oniom_dict:
+                oniom_dict["ecp"] = {}
+            if "ri_scf_basis" not in oniom_dict:
+                oniom_dict["ri_scf_basis"] = None
+            if "ri_cwft_basis" not in oniom_dict:
+                oniom_dict["ri_cwft_basis"] = None
+            if "nprocs" not in oniom_dict:
+                oniom_dict["nprocs"] = 1
+            if "max_memory" not in oniom_dict:
+                oniom_dict["max_memory"] = 1000
+            if oniom_dict["code"] == "mrcc" and "mrcc_calc_inputs" not in oniom_dict:
+                oniom_dict["mrcc_calc_inputs"] = {}
+
+            if oniom_dict["code"] == "orca":
+                if "orca_method_block" not in oniom_dict:
+                    oniom_dict["orca_method_block"] = {"RI": "on", "RunTyp": "Energy"}
+                if "orca_scf_block" not in oniom_dict:
+                    oniom_dict["orca_scf_block"] = {
+                        "HFTyp": "rhf",
+                        "Guess": "PAtom",
+                        "SCFMode": "Direct",
+                        "sthresh": "1e-6",
+                        "AutoTRAHIter": "60",
+                        "MaxIter": "1000",
+                    }
+                if "orca_cation_cap_ecp" not in oniom_dict:
+                    oniom_dict["orca_cation_cap_ecp"] = skzcam_cation_cap_ecp
```

Comment:
Rather than having one big class that's meant to handle ORCA and MRCC, would it not make more sense to have a single base class that you inherit in `recipes/orca/_base.py` and `recipes/mrcc/_base.py` and then modify as needed for that particular code?